File: LeetCode/50. Pow(x, n)/50.java
class Solution {
    public double myPow(double x, int n) {
        if(n == 0) return 1;
        if(n < 0) {
            return 1/x * myPow(1/x, -(n+1));
        }
        double result=1;
        while(n > 0){
            if(n % 2 == 1){
                result *= x;
                --n;
            }
            x *= x;
            n /= 2;
        }
        return result;
    }

        
        
    }

File: LeetCode/50. Pow(x, n)/50.cpp
class Solution {
public:
    double myPow(double x, int n) {
        //Have an if, else to catch positive or negative n
        //square x
        long long N = n;
        if(N < 0){
            x = 1/x;
            N = -N;
        }

        double result = 1.0;
        double current_product = x;

        while(N > 0){
            if (N%2 == 1){
                result *= current_product;
            }
            current_product *= current_product;
            N /= 2;
        }
        return result;
    }
};